### DISCUSSION TITLE
Yaml pipeline implementing different jobs in azure

### DISCUSSION THREAD

**Discussion by untecnologo** [nrwl/nx#23269] [Questions]
### Documentation issue
I have a pipeline with an stage (CodeQualityTestandBuild).
I created the pipeline based on this documentation [https://nx.dev/ci/recipes/set-up/monorepo-ci-azure](https://nx.dev/ci/recipes/set-up/monorepo-ci-azure).

As a part of my strategy I wanted to divide in different jobs my stage:
1 - Job: CodeQuality
2.- Job: Testing
3.- Job: Build

Each one is dependent of the previous job.
Azure DevOps pool, assign a different agent in every job. So at the end of the CodeQuality an artifact (folder with files) should be created to keep the files that run in the pool _temp.

1-. Code Quality : 
- Install the dependencies 
- Set Azure DevOps Organization and Project 
- An run the follow commands
`npm ci`
`npx nx affected --base=$(BASE_SHA) --head=$(HEAD_SHA) -t typecheck`
`npx nx affected --base=$(BASE_SHA) --head=$(HEAD_SHA) -t lint`
`npx nx affected --base=$(BASE_SHA) --head=$(HEAD_SHA) -t prettier-check`
- Then I have to run a task that drop the files in `$(Pipeline.Workspace)/drop` (I think this is mandatory because of the pool per job).

2.- Testing
- Then testing should run the next Task:
`- job: Tests`
    `dependsOn: CodeQuality`
    `steps:`
      `- download: current`
        `artifact: drop`
      `- script: |`
              `cd $(Pipeline.Workspace)/drop/`
              `npx nx affected --base=$(BASE_SHA) --head=$(HEAD_SHA) -t test-coverage`
           
In this moment is when the error happens. I can not have access to the repo to compare because it is not a repo folder.

My question is, is there a way or solution to split in jobs different activities with the Pipeline or it is always mandatory due to the architecture of Azure pipelines that they run together in the same job?

- [ ] Reporting a typo
- [x] Reporting a documentation bug
- [x] Documentation improvement
- [ ] Documentation feedback

### Is there a specific documentation page you are reporting?

[Enter the URL or documentation section here.](https://nx.dev/ci/recipes/set-up/monorepo-ci-azure)

### Additional context or description

Provide any additional details here as needed.

**Comment by FrozenPandaz** [comment]
I'm going to convert this to a discussion question so that it can be discussed more easily.

Please also consider joining the [Community Discord](https://go.nx.dev/community) and start a discussion there.
